crates/trie/src/trie.rs

```diff
@@ -438,13 +405,13 @@ where
     /// # Returns
     ///
     /// The storage root.
-    pub fn root(&self) -> Result<B256, StorageRootError> {
+    pub fn root(self) -> Result<B256, StorageRootError> {
```

Comment:
why this change?

Reply:
oops, did not push [this follow up](https://github.com/paradigmxyz/reth/pull/6502/commits/a7199b5e82fb6ec0246d2c0f103006348a9d6efa). allows us to get rid of unnecessary clone since `StorageRoot` calculators are short live anyways. this mirrors the interface of `StateRoot`

Reply:
gotcha